runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5574,6 +5574,46 @@ void do_nic_amo_nf_V(int v_len, uint64_t* opnd1_v, c_nodeid_t* locale_v,
 }
 
 
+void chpl_comm_getput_unordered(c_nodeid_t dst_locale, void* dst_addr,
+                                c_nodeid_t src_locale, void* src_addr,
+                                size_t size, int32_t typeIndex,
+                                int32_t commID, int ln, int32_t fn)
+{
+  assert(dst_addr != NULL);
+  assert(src_addr != NULL);
+
+  if (size == 0)
+    return;
+
+  if (dst_locale == chpl_nodeID && src_locale == chpl_nodeID) {
+    memmove(dst_addr, src_addr, size);
+    return;
+  }
+
+  if (dst_locale == chpl_nodeID) {
+    chpl_comm_get_unordered(dst_addr, src_locale, src_addr, size, typeIndex, commID, ln, fn);
+  } else if (src_locale == chpl_nodeID) {
+    // TODO want chpl_comm_put_unordered
+    chpl_comm_put(src_addr, dst_locale, dst_addr, size, typeIndex, commID, ln, fn);
+  } else {
+    // TODO use unordered ops in this case? Would have to ensure we always
+    // flush GET buffer before the PUT buffer
+    if (size <= 1024) {
```

Comment:
`#define` this constant just once and use the defined value in both places, so that there's zero chance they get out of sync.